### PR TITLE
Adding support for newer versions of MSVC

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -76,7 +76,7 @@ set(VEXCL_BOOST_COMPONENTS
     program_options
     CACHE STRING "Boost components required by VexCL.")
 
-if (MSVC10)
+if (MSVC10 OR MSVC11 OR MSVC12 OR MSVC14)
```
Comment (ddemidov) [inline CMakeLists.txt]: Do you know if there is more general way to detect MSVC?

**Comment by ddemidov** [comment]
I only use boost.chrono when [`_MSC_VER < 1700`](https://github.com/ddemidov/vexcl/blob/0411cbdc86b1efa8d33b645a82990ac09a8e60b1/vexcl/profiler.hpp#L47-L50) (later versions of MSVC do provide std::chrono). I think that was the reason for only linking to boost.chrono for MSVC10. I'll merge it if this does not work for as is.

**Comment by hkaiser** [comment]
It did not work for me using VS2015, it was complaining about not being able to find libboost_chrono.

File: CMakeLists.txt (hunk 1 above)
Comment (hkaiser) [inline CMakeLists.txt]: `if(MSVC)` should do the trick

Comment (ddemidov) [inline CMakeLists.txt]: Would it make sense to replace the condition then?

**Comment by ddemidov** [comment]
Thanks, I've merged this and also did 8a65534e8b515aa3b71e592c7d78e97ef991d489.

**Comment by hkaiser** [comment]
Thanks, works fine now.

**Pull request merged**